### PR TITLE
[Bugfix]refresh after drop partition with data

### LINKED ISSUE #9987: [sync materialized view]did not refresh after drop partition with data
### Steps to reproduce the behavior (Required)
```
CREATE TABLE `t1` (
                                    `k1` date,
                                    `k2` int,
                                    `k3` int
                                ) ENGINE=OLAP
                                DUPLICATE KEY(`k1`)
                                COMMENT "OLAP"
                                PARTITION BY RANGE (k1) (
                                PARTITION `p202006` VALUES LESS THAN ("2020-07-01"),
                                PARTITION `p202007` VALUES LESS THAN ("2020-08-01"),
                                PARTITION `p202008` VALUES LESS THAN ("2020-09-01")
                                )
                                DISTRIBUTED BY HASH(`k1`) BUCKETS 3;
INSERT INTO t1 VALUES ("2020-07-02",1,1);
CREATE MATERIALIZED VIEW mv1
                             PARTITION BY k1
                             DISTRIBUTED BY HASH(k1) BUCKETS 10
                             REFRESH ASYNC
                             AS SELECT k1,k2 FROM t1;
ALTER TABLE t1 DROP PARTITION p202007;
SELECT * FROM mv1;
```

### Expected behavior (Required)
`Empty set (0.01 sec)`
### Real behavior (Required)
```
+------------+------+
| k1         | k2   |
+------------+------+
| 2020-07-02 |    1 |
+------------+------+
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.4-RELEASE 9744f36 |
+----------------------------+
```

### PR DESCRIPTION
**Pull request by smartlxh**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9987

## Problem Summary(Required) ：

did not refresh after drop partition with data
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 7 (71.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|7|71.43%|[1443, 1444]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 9 (55.56%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|9|55.56%|[1442, 1443, 1446, 1447]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -1435,6 +1435,17 @@ public void dropPartition(Database db, Table table, DropPartitionClause clause)
                 }
             }
             tabletIdSet = olapTable.dropPartition(db.getId(), partitionName, clause.isForceDrop());
+            try {
+                for (Long mvId : olapTable.getRelatedMaterializedViews()) {
+                    MaterializedView materializedView = (MaterializedView) db.getTable(mvId);
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1440]: materializedView may got null better to check this

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -1435,6 +1435,17 @@ public void dropPartition(Database db, Table table, DropPartitionClause clause)
...
+                    if (materializedView.isLoadTriggeredRefresh()) {
+                        GlobalStateMgr.getCurrentState().getLocalMetastore().refreshMaterializedView(
+                                db.getFullName(), materializedView.getName(), Constants.TaskRunPriority.NORMAL.value());
+                    }
+                }
+            } catch (MetaNotFoundException e) {
+                throw new DdlException(e.getMessage());
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1447]:
```suggestion
                throw new DdlException(“fail to refresh materialized views when dropping partition”, e);
```

**Review by ABingHuang — CHANGES_REQUESTED** [review]

**Comment by smartlxh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 9 (55.56%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|9|55.56%|[1442, 1443, 1446, 1447]|

**Review by Astralidea — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Review by decster — APPROVED** [review]

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|3|3|100.00%|[]|

**Pull request merged**